exp/api/v1beta1/azuremanagedmachinepool_webhook_test.go

```diff
@@ -357,6 +357,90 @@ func TestAzureManagedMachinePoolUpdatingWebhook(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "Unchanged EnabledUltraSSD in an agentpool should not result in an error",
+			new: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					EnableUltraSSD: to.BoolPtr(true),
+				},
+			},
+			old: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					EnableUltraSSD: to.BoolPtr(true),
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Unchanged EnabledUltraSSD in an agentpool should not result in an error",
+			new: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					EnableUltraSSD: to.BoolPtr(false),
+				},
+			},
+			old: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					EnableUltraSSD: to.BoolPtr(false),
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "EnableUltraSSD feature is immutable",
+			new: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					EnableUltraSSD: to.BoolPtr(false),
+				},
+			},
+			old: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					EnableUltraSSD: to.BoolPtr(true),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "EnableUltraSSD feature is immutable",
```

Comment:
nit: it would be nice if these names were unique so it's easier to debug if they break in the future

Reply:
Made them more descriptive to the logic applied